tests/tracing_test.go

```diff
@@ -158,6 +161,137 @@ func TestTracing(t *testing.T) {
+	}
+	// Start test server
+	ts := httptest.NewServer(http.HandlerFunc(
+		func(w http.ResponseWriter, r *http.Request) {
+			fmt.Fprint(w, html)
+		},
+	))
+	defer ts.Close()
+
+	// Initialize VU and browser module
+	vu := k6test.NewVU(t, k6test.WithTracerProvider(tp))
+
+	rt := vu.Runtime()
+	root := browser.New()
+	mod := root.NewModuleInstance(vu)
+	jsMod, ok := mod.Exports().Default.(*browser.JSModule)
+	require.Truef(t, ok, "unexpected default mod export type %T", mod.Exports().Default)
+	require.NoError(t, rt.Set("browser", jsMod.Browser))
+	vu.ActivateVU()
+
+	testCases := []struct {
+		name     string
+		js       string
+		expected []string
+	}{
+		{
+			name: "goto",
+			js: fmt.Sprintf(`
+				page = await browser.newPage();
+				await page.goto('%s');
+				browser.closeContext();
+				`, ts.URL),
+			expected: []string{
+				"iteration",
+				"browser.newPage",
+				"browser.newContext",
+				"browserContext.newPage",
+				"navigation", // created when a new page is created
+				"page.goto",
+				"navigation", // created when a navigation occurs after goto
+			},
+		},
+		{
+			name: "reload",
+			js: fmt.Sprintf(`
+				page = await browser.newPage();
+				await page.goto('%s');
+				await page.reload();
+				browser.closeContext();
+				`, ts.URL),
+			expected: []string{
+				"iteration",
+				"browser.newPage",
+				"browser.newContext",
+				"browserContext.newPage",
+				"navigation", // created when a new page is created
+				"page.goto",
+				"navigation", // created when a navigation occurs after goto
+				"page.reload",
+				"navigation", // created when a navigation occurs after reload
+			},
+		},
+		{
+			name: "go_back",
+			js: fmt.Sprintf(`
+				page = await browser.newPage();
+				await page.goto('%s');
+				await page.evaluate(() => window.history.back());
+				browser.closeContext();
+				`, ts.URL),
+			expected: []string{
+				"iteration",
+				"browser.newPage",
+				"browser.newContext",
+				"browserContext.newPage",
+				"navigation", // created when a new page is created
+				"page.goto",
+				"navigation", // created when a navigation occurs after goto
+				"navigation", // created when going back to the previous page
+			},
+		},
+		{
+			name: "same_page_navigation",
+			js: fmt.Sprintf(`
+				page = await browser.newPage();
+				await page.goto('%s');
+				await page.locator('a[id=\"top\"]').click();
+				browser.closeContext();
+				`, ts.URL),
+			expected: []string{
+				"iteration",
+				"browser.newPage",
+				"browser.newContext",
+				"browserContext.newPage",
+				"navigation", // created when a new page is created
+				"page.goto",
+				"navigation", // created when a navigation occurs after goto
+				"locator.click",
+				"navigation", // created when navigating within the same page
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		// Cannot create new VUs that do not depend on each other due to the
+		// sync.Once in mod.NewModuleInstance, so we can't parallelize these
+		// subtests.
```

Comment:
Why don't we use [the `startIteration` helper](https://github.com/grafana/xk6-browser/blob/b8d641810cbc8cdd9cb1c90a93592909e61cc860/tests/helpers.go#L13-L41) and run tests in parallel as in here:

https://github.com/grafana/xk6-browser/blob/b8d641810cbc8cdd9cb1c90a93592909e61cc860/tests/page_test.go#L185-L244

Reply:
Hmm, yeah, that's a good shout. I'll do it in another PR as it might take sometime to refactor the test.

Reply:
Sure: https://github.com/grafana/xk6-browser/issues/1425

It's straightforward to create an issue on GH from a comment now 🎉